cli/src/main.rs

```diff
@@ -0,0 +1,52 @@
+use std::ops::Deref;
+
+use clap::{Parser, Subcommand};
+
+use color_eyre::Result;
+
+/// Shell CLI
+#[derive(Debug, Parser)]
+#[clap(version, about)]
+struct Cli {
+    /// Device
+    #[clap(short, long, required = true,  value_parser = clap::builder::NonEmptyStringValueParser::new())]
+    device: String, // not an Option<String> because the device id/name is required
+
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+// these commands canbe called from the CLI using lowercase Commands name
+#[derive(Subcommand, Debug)]
+enum Commands {
+    /// Execute a command
+    Command { cmd: String },
+    // Authenticate device
+    // Authenticate { token: String },
+    // Logout device
+    // Logout,
+}
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+
+    let cli = Cli::parse();
+
+    match &cli.command {
+        Commands::Command { cmd } => {
+            println!("Input command \"{}\"", cmd); // substitute with logging inside the function
+
+            // parse the cmd into a slice
+            let cmd = shellwords::split(cmd.trim())
+                .or(Err(rust_remote_shell::ShellError::MalformedInput))?;
```

Comment:
Here you should map_err for the malformed input because the [MismatchedQuotes](https://docs.rs/shellwords/latest/shellwords/struct.MismatchedQuotes.html) information is lost